contracts/src/StabilityPool.sol

```diff
@@ -524,8 +528,14 @@ contract StabilityPool is LiquityBase, IStabilityPool, IStabilityPoolEvents {
             emit ScaleUpdated(currentScale);
             // If there's no scale change and no pool-emptying, just do a standard multiplication
         } else {
-            newP = currentP * newProductFactor / DECIMAL_PRECISION;
+            uint256 errorFactor;
+            if (lastBoldLossError_Offset > 0) {
+                errorFactor = lastBoldLossError_Offset * newProductFactor / lastBoldLossError_TotalDeposits;
```

Comment:
Trying to understand the rationale for the denominator here.

`lastBoldLossError_TotalDeposits` is only assigned a value in L458 (for a pool-emptying liq) and in L538 after it's been used in the division in L533.

So that means 1) if the current liq is pool-emptying, we use current  `_totalBoldDeposits` in the division. But if 2) it's not pool emptying, we use the stored value `lastBoldLossError_TotalDeposits` which is stored at the previous liq.

Is that intended? If so, why? 

In case 1), we use the most recent totalBoldDeposits, which could have changed since the last liq as deposits were added/removed.  In 2, we use a potentially outdated stored value.

I'd have assumed we should always use the current value, since we're scaling a past error (calculated using past total deposits) by current total deposits.

Reply:
> Is that intended? If so, why?

Yes, that’s intended. In case 1 we are kind of resetting everything, `P` to 1 too. So we need to reset the error tracking too.

Anyway, there’s a TODO in this PR to review the case for scale updates, there may be something missing (although as far I know, it hasn’t surfaced during fuzzing yet). So, it’s intended, but I’m not 100% sure it’s correct. I want to get deep into that case once we make sure the basic logic (for the generic case where `P` is not reset) is sound.